1702 document feat introduce alter parallelism (#1748)
risingwavelabs/risingwave-docs: docs/concepts/key-concepts.md

  <link rel="canonical" href="https://docs.risingwave.com/docs/current/key-concepts/" />
</head>

> This page explains key concepts and terms that are used throughout the documentation. It is divided into two parts. In the first part, we present some terms strongly related to RisingWave. In the second part, we includ a broader range of terms in a glossary. 

Reviewer: ```suggestion
This page explains key concepts and terms that are used throughout the documentation. In the first part, we present terms closely related to RisingWave. In the second part, we include a broader range of terms in a glossary. 
```

## Key concepts of RisingWave

### Clusters

> A group of interconnected nodes and services that acts as a single system running an instance of RisingWave.

Reviewer 2: act, not acts

### Compute nodes

The compute nodes in RisingWave are responsible for ingesting data from upstream systems, parsing and running SQL queries, and delivering data to downstream systems.

> ### Compactor nodes

Reviewer 2: Compact nodes

The compact nodes handle data storage and retrieval from object storage. They also perform data compaction to optimize storage efficiency.

### Fragments

> In RisingWave, When a streaming query plan executes, it divides into multiple independent fragments to allow parallel execution. Each fragment is a chain of SQL operators. Under the hood, it is executed by parallel actors. The degree of parallelism between fragments can be different.

Reviewer 2: In RisingWave, **when** a streaming query plan executes (lower case)

### Indexes

> Indexes in a database are typically created on one or more columns of a table, allowing the database management system (DBMS) to  locate and retrieve the desired data from the table quickly. This can greatly improve the performance of database queries, especially for large tables or frequently accessed tables.

Reviewer: ```suggestion
Indexes in a database are typically created on one or more columns of a table, allowing the database management system (DBMS) to locate and retrieve the desired data from the table quickly. This can significantly improve the performance of database queries, especially for large tables or frequently accessed tables. In RisingWave, indexes can speed up batch queries.
```

### Materialized views

When the results of a view expression are stored in a database system, they are called materialized views. In RisingWave, the result of a materialized view is updated when a relevant event arrives in the system. When you query the result, it is returned instantly as the computation has already been completed when the data comes in. You need to use the [`CREATE MATERIALIZED VIEW`](/sql/commands/sql-create-mv.md) statement to create a materialized view.

### Meta node

The meta node takes charge of managing the metadata of compute and compact nodes and orchestrating operations across the system.

### Nodes

> A node is a logical collection of IT resources that handles specific workloads based on their types. There are two types of nodes in RisingWave:

Author: @CharlieSYH  Could you pls check this concept? It lists 3 nodes but says it has two types of nodes. I'm a bit confused.

- Compute node

> - Compactor node

Reviewer 2: Compact node

- Meta node

### Parallelism

> Parallelism refers to the technique of simultaneously executing multiple database operations or queries to improve performance and increase efficiency. It involves dividing a database workload into smaller tasks and executing them concurrently on multiple processors or machines. In RisingWave, you can set the parallelism of streaming jobs, like [tables](../sql/commands/sql-alter-table.md#set-parallelism), [materialized views](../sql/commands/sql-alter-materialized-view.md#set-parallelism), and [sinks](../sql/commands/sql-alter-sink.md#set-parallelism).

Reviewer: ```suggestion
Parallelism refers to the technique of simultaneously executing multiple database operations or queries to improve performance and increase efficiency. It involves dividing a database workload into smaller tasks and executing them concurrently on multiple processors or machines. In RisingWave, you can set the parallelism of streaming jobs, like [tables](/sql/commands/sql-alter-table.md#set-parallelism), [materialized views](/sql/commands/sql-alter-materialized-view.md#set-parallelism), and [sinks](/sql/commands/sql-alter-sink.md#set-parallelism).
```

Reviewer: We use absolute paths (starting after `/docs`), not relative paths, in our docs. See this page: https://www.notion.so/risingwave-labs/Tips-for-Markdown-Docusaurus-70bc8f0c686f4fdf82430eba1fd582fb?pvs=4#619b078bf36046b5aa04d3ecb381701b
